synapse/events/snapshot.py

```diff
@@ -77,16 +78,30 @@ def __init__(self):
 
         self.app_service = None
 
-    def serialize(self):
+    def serialize(self, event):
         """Converts self to a type that can be serialized as JSON, and then
         deserialized by `deserialize`
 
+        Args:
+            event (FrozenEvent): The event that this context relates to
+
         Returns:
             dict
         """
+
+        # We don't serialize the full state dicts, instead they get pulled out
+        # of the DB on the other side. However, the other side can't figure out
+        # the prev_state_ids, so if we're a state event we include the event
+        # id that we replaced in the state.
+        if event.is_state():
```

Comment:
given `event.is_state()` does two lookups of `state_key`, leading to a total of six lookups in this function, I wonder if we could/should do `getattr(event, 'state_key')` once instead.

probably overkill though.

Reply:
Ha, I hadn't noticed. I don't think it really matters for now though